ci-operator/step-registry/cucushift/installer/rehearse/aws/cases/all-regions/provision/cucushift-installer-rehearse-aws-cases-all-regions-provision-commands.sh

```diff
@@ -0,0 +1,534 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+trap 'post_step_actions' EXIT TERM INT
+
+if [[ -z "$OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE" ]]; then
+  echo "OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE is an empty string, exiting"
+  exit 1
+fi
+echo "Installing from release ${OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE}"
+export AWS_SHARED_CREDENTIALS_FILE="${CLUSTER_PROFILE_DIR}/.awscred"
+
+INSTALL_BASE_DIR=/tmp/install_base_dir
+mkdir -p ${INSTALL_BASE_DIR}
+ssh_pub_key=$(<"${CLUSTER_PROFILE_DIR}/ssh-publickey")
+pull_secret=$(<"${CLUSTER_PROFILE_DIR}/pull-secret")
+ret=0
+
+# following regions will be tested
+REGIONS_LIST=${ARTIFACT_DIR}/regions.txt
+RESULT=${SHARED_DIR}/result.json
+echo '{}' > ${RESULT}
+
+
+function post_step_actions()
+{
+  set +o errexit
+  pushd $INSTALL_BASE_DIR
+  # find . -name metadata.json -exec cp --parents '{}' ${SHARED_DIR}  \;
+  cp ${RESULT} ${ARTIFACT_DIR}/
+  find . -name "log-bundle-*.tar.gz" -exec cp --parents '{}' ${ARTIFACT_DIR}  \;
+
+  find . -name .openshift_install.log -exec cp --parents '{}' ${ARTIFACT_DIR}  \;
+  find ${ARTIFACT_DIR} -name .openshift_install.log -exec sed -i 's/password: .*/password: REDACTED/; s/X-Auth-Token.*/X-Auth-Token REDACTED/; s/UserData:.*,/UserData: REDACTED,/;' '{}' \;
+  popd
+  set -o errexit
+
+  echo "--- ARTIFACT_DIR ---"
+  find ${ARTIFACT_DIR} -type f
+  echo "--- SHARED_DIR ---"
+  find ${SHARED_DIR} -type f
+  echo "--- INSTALL_BASE_DIR ---"
+  find ${INSTALL_BASE_DIR} -type f
+  echo "--- RESULTS ---"
+  echo -e "region\tcluster_name\tinfra_id\tinstall\thealth_check"
+  jq -r '.[] | [.region, .cluster_name, .infra_id, .install_result, .health_check_result] | @tsv' $RESULT
+}
+
+
+function generate_regions()
+{
+  if [[ ${CUSTOM_REGION_LIST} == "" ]]; then
+    # Get all regions
+    aws --region $LEASED_RESOURCE ec2 describe-regions > ${ARTIFACT_DIR}/all_regions.json
+    jq -r '.Regions[] | select(.OptInStatus=="opted-in" or .OptInStatus=="opt-in-not-required") | .RegionName' ${ARTIFACT_DIR}/all_regions.json | sort > ${REGIONS_LIST}
+
+    if [[ ${REGIONS_IGNORED} != "" ]]; then
+      echo "Following regions will be ignored: ${REGIONS_IGNORED}"
+      for r in $REGIONS_IGNORED;
+      do
+        sed -i "s/${r}//g" $REGIONS_LIST
+      done
+      sed -i '/^[[:space:]]*$/d' $REGIONS_LIST
```

Comment:
I think the two sed lines can be merged as `sed -i "/^${r}$/d" $REGIONS_LIST`

Reply:
Fixed.